### PR TITLE
Fix heisenbug dec 2023

### PR DESCRIPTION
**Pull request by MatthiasHeilManchester**
- Work-around for bizarre MPI problem in distributed triangle meshes. Code used to 
seg fault when going into MPI_Wait if  oomph-lib was compiled with any 
optimisation, i.e. -O1 upwards.   ddd-based investigations unclear. Trial and error showed that
 passing a new (rather than re-used) MPI_Request object to that
 call (and the associated collective communications that it's
 waiting for) fixed the problem. Somewhat unsatisfactory and definitely
 one to keep an eye on...

ANDREW: Can you please clone my branch and check if it works on your machine too (either pre- or post-accept for this merge)

- A few more little tidies and modified the build machinery to explicity suppress Qt5-related builds in CGAL.

- Ubuntu passes cleanly

     https://github.com/MatthiasHeilManchester/oomph-lib/actions/runs/7346957076

- macos still has the usual failure in the elastic bretherton code

     https://github.com/MatthiasHeilManchester/oomph-lib/actions/runs/7346957077

### CODE REVIEW

File: src/meshes/triangle_mesh.template.cc [hunk 1]
```diff
@@ -34560,10 +34566,6 @@ namespace oomph
         this->copy_connection_information(polygon_pt->polyline_pt(p),
                                           tmp_curve_section_pt);
 
-        // Now update the polyline according to the new vertices but
```
Comment (alhazel) [inline src/meshes/triangle_mesh.template.cc:34563]: Is this now redundant? Looks a bit dangerous to remove, unless the potential memory leak is addressed elsewhere.

**Comment by alhazel** [comment]
I have confirmed that the branch works on my machine under both OpenMPI and mpich. Flags in the validate script need to be changed for Mpich are to allow --output-filename (OpenMPI) and --outfile-pattern (mpich).  Once this change is included then I'm happy to approve the request.

File: src/meshes/triangle_mesh.template.cc (hunk 1 above)
Comment (MatthiasHeilManchester) [reply src/meshes/triangle_mesh.template.cc:34563]: not yet! I've just pushed the change to the mpirun flag. May as well wait until the github ci tests pass for that too.

**Comment by MatthiasHeilManchester** [comment]
> I have confirmed that the branch works on my machine under both OpenMPI and mpich. Flags in the validate script need to be changed for Mpich are to allow --output-filename (OpenMPI) and --outfile-pattern (mpich). Once this change is included then I'm happy to approve the request.

Done. All self tests pass on github:

- https://github.com/MatthiasHeilManchester/oomph-lib/actions/runs/7643357264
- https://github.com/MatthiasHeilManchester/oomph-lib/actions/runs/7643357256

File: src/meshes/triangle_mesh.template.cc (hunk 1 above)
Comment (alhazel) [reply src/meshes/triangle_mesh.template.cc:34563]: OK. All Approved.

**Pull request merged**